### ISSUE TITLE
Upgrade DB schema for courses

### ISSUE THREAD

**Issue by hmil** [CourseAdvisor/courseadvisor#106]
Make sure the DB schema can handle edge cases described in #105

**Comment by hmil** [comment]
One way to do it would be:
- Extract course attributes that are bound to a course instance in a separate table
- Add a year / term couple to the review table (ask the student when he took the course, prefill this field if we know this already from the inscriptions)

```
course:                            review:
- name                             - grade
- string_id                   <==  - year
- grade_avg (cached average)       - term

    /\
    ||

course_inst:
- year
- term
- credits
- teacher
```

**Comment by hmil** [comment]
Closed with #108
